pytensor/sparse/basic.py

```diff
@@ -4260,3 +4256,85 @@ def grad(self, inputs, grads):
 
 
 construct_sparse_from_list = ConstructSparseFromList()
+
+
+class SparseBlockDiagonalMatrix(BaseBlockDiagonal):
+    def make_node(self, *matrices, format: Literal["csc", "csr"] = "csc", name=None):
+        if not matrices:
+            raise ValueError("no matrices to allocate")
+        dtype = largest_common_dtype(matrices)
+        matrices = list(map(as_sparse_or_tensor_variable, matrices))
+
+        if any(mat.type.ndim != 2 for mat in matrices):
+            raise TypeError("all data arguments must be matrices")
+
+        out_type = matrix(format=format, dtype=dtype, name=name)
+        return Apply(self, matrices, [out_type])
+
+    def perform(self, node, inputs, output_storage, params=None):
+        format = node.outputs[0].type.format
```

Comment:
After you make format a prop:

```suggestion
        format = self.format
```